org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/Filters.java

```diff
@@ -15,54 +15,70 @@
 import org.objectweb.asm.tree.MethodNode;
 
 /**
- * Filter that combines other filters.
+ * Factory for all JaCoCo filters.
  */
-public final class Filters implements IFilter {
+public final class Filters {
 
 	/**
 	 * Filter that does nothing.
 	 */
-	public static final IFilter NONE = new Filters();
-
-	private final IFilter[] filters;
+	public static final IFilter NONE = new FilterSet();
 
 	/**
-	 * Creates filter that combines all other filters.
+	 * Creates a filter that combines all filters.
 	 *
-	 * @return filter that combines all other filters
+	 * @return filter that combines all filters
 	 */
 	public static IFilter all() {
-		return new Filters(new EnumFilter(), new SyntheticFilter(),
-				new BridgeFilter(), new SynchronizedFilter(),
-				new TryWithResourcesJavac11Filter(),
-				new TryWithResourcesJavacFilter(),
-				new TryWithResourcesEcjFilter(), new FinallyFilter(),
-				new PrivateEmptyNoArgConstructorFilter(), new AssertFilter(),
-				new StringSwitchJavacFilter(), new StringSwitchFilter(),
-				new EnumEmptyConstructorFilter(), new RecordsFilter(),
-				new ExhaustiveSwitchFilter(), //
-				new RecordPatternFilter(), //
-				new AnnotationGeneratedFilter(), new KotlinGeneratedFilter(),
-				new KotlinEnumFilter(), //
-				new KotlinLateinitFilter(), new KotlinWhenFilter(),
-				new KotlinWhenStringFilter(),
-				new KotlinUnsafeCastOperatorFilter(),
-				new KotlinNotNullOperatorFilter(),
-				new KotlinInlineClassFilter(),
-				new KotlinDefaultArgumentsFilter(), new KotlinInlineFilter(),
-				new KotlinCoroutineFilter(), new KotlinDefaultMethodsFilter(),
-				new KotlinComposeFilter());
+		return new FilterSet( //
+				allJavaFilters(), //
+				allKotlinFilters());
 	}
 
-	private Filters(final IFilter... filters) {
-		this.filters = filters;
+	private static IFilter allJavaFilters() {
```

Comment:
There are applied even if source language is not Java, so I prefer

```suggestion
	private static IFilter allCommonFilters() {
```